openscm/core.py

```diff
@@ -1,24 +1,304 @@
 """
-Then OpenSCM low-level API includes the basic functionality to run a
-particular simple climate model with OpenSCM as well as
-setting/getting its parameter values. Mapping of parameter names and
-units is done internally.
+The OpenSCM low-level API includes the basic functionality to run a particular
+simple climate model with OpenSCM as well as setting/getting its parameter values.
+Mapping of parameter names and units is done internally.
 
-Parts of this API definition seems unpythonic as it is designed to
-be easily implementable in several programming languages.
+Parts of this API definition seems unpythonic as it is designed to be easily
+implementable in several programming languages.
 """
 
-from typing import Sequence, Tuple
+from enum import Enum
+from typing import Any, Dict, Sequence, Tuple
 
 
-class ParameterLengthError(Exception):
+class ParameterType(Enum):
+    """
+    Parameter type.
+    """
+
+    SCALAR = 1
+    TIMESERIES = 2
+
+
+class _Parameter:
+    """
+    Represents a parameter in the parameter hierarchy.
```

Comment:
We need to add some explanation of what we mean by the term 'Parameter' here

Reply:
Good point. I have a section in the docs for the low-level API. Repeat here?

Reply:
reference here (you can cross reference in docstrings)

Reply:
done